plugins/module_utils/ansible_freeipa_module.py

```diff
@@ -616,7 +616,7 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         # TODO: shouldn't we also disconnect from api backend?
         temp_kdestroy(self.ccache_dir, self.ccache_name)
 
-        self.exit_json(changed=self.changed, user=self.exit_args)
+        self.exit_json(changed=self.changed, **self.exit_args)
```

Comment:
This should be `dnszone=self.exit_args` to make it compatible to other modules.

Reply:
It should. And it should be also fixed on `ipavault`.

Reply:
This will not be changed here.

As this code should be usable by other modules, currently, the module itself is responsible of handling the result dictionary.